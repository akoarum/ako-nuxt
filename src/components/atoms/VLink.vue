<template>
  <a class="vLink">
    <VCaret class="vLink__arrow" />
    <slot />
  </a>
</template>

<script lang="ts">
import Vue from 'vue'
import { VCaret } from '~/components/icons'

export default Vue.extend({
  components: { VCaret }
})
</script>

<style lang="scss" scoped>
.vLink {
  position: relative;
  display: inline-block;
  text-decoration: none;

  @include media-min {
    padding-left: 12px;
    font-size: $fontSize-pc;
  }

  @include media-max {
    padding-left: 12px;
    font-size: $fontSize-sp;
  }
}
.vLink__arrow {
  position: absolute;
  left: 0;

  @include media-min {
    top: 0.35em;
  }

  @include media-max {
    top: 0.3em;
  }
}
</style>
