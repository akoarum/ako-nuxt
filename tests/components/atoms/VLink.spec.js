import { shallowMount } from '@vue/test-utils'
import VLink from '~/components/atoms/VLink'

describe('VLink', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VLink, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
