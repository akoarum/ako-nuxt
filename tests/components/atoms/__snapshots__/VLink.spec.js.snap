// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VLink initialized snapshot 1`] = `
<a
  class="vLink"
>
  <vcaret-stub
    class="vLink__arrow"
  />
   テスト
</a>
`;
